app/(auth)/signup/components/RegisterForm.tsx

```diff
@@ -0,0 +1,139 @@
+"use client";
+
+import { useState } from "react";
+import { useForm, FormProvider } from "react-hook-form";
+import { z } from "zod";
+import { authSchema } from "../../../../src/schema/schemas";
+import Profile from "../../../../components/Profile";
+import User from "../../../../components/User";
+import { Button } from "@/components/ui/button";
+import { signup } from "@/src/lib/auth-actions";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation } from "@tanstack/react-query";
+
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Progress } from "@/components/ui/progress";
+
+const steps = ["user", "profile"];
+type input = z.infer<typeof authSchema>;
+export default function RegisterForm() {
+  const [stepIndex, setStepIndex] = useState(0);
+  const totalSteps = steps.length;
+  const progressValue = (stepIndex / (totalSteps - 1)) * 100;
+
+  const form = useForm<input>({
+    resolver: zodResolver(authSchema),
+    mode: "onChange",
+    defaultValues: {
+      username: "",
+      name: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+      favoriteType: "Water",
+      favoritePokemon: "",
+      avatar: "",
+    },
+  });
+
+  const handleNext = async () => {
+    let valid = false;
+
+    if (stepIndex === 0) {
+      valid = await form.trigger([
+        "name",
+        "email",
+        "username",
+        "password",
+        "confirmPassword",
+      ]);
+      console.log(valid);
+    } else if (stepIndex === 1) {
+      valid = await form.trigger(["favoriteType", "favoritePokemon", "avatar"]);
+      console.log(valid);
+    }
+
+    if (valid) {
+      setStepIndex((prev) => prev + 1);
+    }
+  };
+  const handlePrevious = () => setStepIndex((prev) => prev - 1);
+
+  const onSubmit = async (data: input) => {
+    console.log("I am called yay");
+    const finalData = form.getValues();
+
+    console.log("Final Form Data:", finalData);
+    const formDataObj = new FormData();
+    Object.entries(finalData).forEach(([key, value]) => {
+      formDataObj.append(key, value);
+    });
+    formDataObj.forEach((value, key) => {
+      console.log(key, value);
+    });
+
+    await signup(formDataObj);
+  };
```

Comment:
```suggestion
  const onSubmit = async (data: input) => {
    console.log("I am called yay");
    const finalData = form.getValues();

  

    await signup(finalData);
  };
```

Just pass in the data directly. Also you can just directly use the data argument of the function